docs/pages/guides/preloading-and-caching-assets.md

```diff
@@ -60,45 +57,65 @@ function cacheFonts(fonts) {
   return fonts.map(font => Font.loadAsync(font));
 }
 
-export default class AppContainer extends React.Component {
-  state = {
-    isReady: false,
-  };
-
-  async _loadAssetsAsync() {
-    const imageAssets = cacheImages([
-      'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png',
-      require('./assets/images/circle.jpg'),
-    ]);
+export default function App() {
+  const [appIsReady, setAppIsReady] = useState(false);
+
+  // Load any resources or data that you need prior to rendering the app
+  useEffect(() => {
+    async function loadResourcesAndDataAsync() {
+      try {
+        SplashScreen.preventAutoHideAsync();
+
+        const imageAssets = cacheImages([
+          'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png',
+          require('./assets/images/circle.jpg'),
+        ]);
+
+        const fontAssets = cacheFonts([FontAwesome.font]);
+
+        await Promise.all([...imageAssets, ...fontAssets]);
+      } catch (e) {
+        // You might want to provide this error information to an error reporting service
+        console.warn(e);
+      } finally {
+        setAppIsReady(true);
+        SplashScreen.hideAsync();
+      }
+    }
 
-    const fontAssets = cacheFonts([FontAwesome.font]);
+    loadResourcesAndDataAsync();
+  }, []);
 
-    await Promise.all([...imageAssets, ...fontAssets]);
+  if (!appIsReady) {
+    return null;
   }
 
-  render() {
-    if (!this.state.isReady) {
-      return (
-        <AppLoading
-          startAsync={this._loadAssetsAsync}
-          onFinish={() => this.setState({ isReady: true })}
-          onError={console.warn}
-        />
-      );
-    }
-
-    return (
-      <View>
-        <Text>Hello world, this is my app.</Text>
-      </View>
-    );
-  }
+  return (
+    <View style={styles.container}>
+      <Text>Hello world, this is my app.</Text>
+    </View>
+  );
 }
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#fff',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+});
 ```
 
 </SnackInline>
 
-See a full working example in [this Expo template project](https://github.com/expo/expo/blob/sdk-43/templates/expo-template-tabs/App.tsx). You can also run `npx create-expo-app --template tabs`, which will set you up locally with the same template.
+To use the local image asset, you can continue referencing the source of the image normally in your project, for example:
+
+```jsx
+<Image source={require('path/to/image.png')} />
+```
+
+See the complete working example in [Expo's tabs template project](https://github.com/expo/expo/blob/sdk-43/templates/expo-template-tabs/App.tsx). You can also run `npx create-expo-app --template tabs` to set up a local project with the same template.
```

Comment:
```suggestion
See the complete working example in [Expo's tabs template project](https://github.com/expo/expo/blob/main/templates/expo-template-tabs/App.tsx). You can also run `npx create-expo-app --template tabs` to set up a local project with the same template.
```

Using `main` is also not ideal, but I feel like it's bit better than linking specific SDK branch, which in time will need to updated.